### PR TITLE
[SDK refactoring #5] MetaDataApi refactoring

### PR DESCRIPTION
**Pull request by shoom3301**
Docs: https://www.notion.so/cownation/FE-SDK-Dogfooding-our-own-SDK-in-CoW-Swap-ce502a2dfb3f423bb9c7998de4237575
Swagger: https://github.com/cowprotocol/services/pull/1208
Explorer dogfooding: https://github.com/cowprotocol/explorer/pull/352
General refactoring draft: https://github.com/cowprotocol/cow-sdk/pull/96

## What changed
1. Refactored MetaDataApi (nothing special, just removed the context)

### CODE REVIEW

File: src/common/chains.ts [hunk 1]
```diff
@@ -3,9 +3,3 @@ export enum SupportedChainId {
   GOERLI = 5,
   GNOSIS_CHAIN = 100,
 }
-
-export const ALL_SUPPORTED_CHAIN_IDS: SupportedChainId[] = [
-  SupportedChainId.MAINNET,
-  SupportedChainId.GOERLI,
-  SupportedChainId.GNOSIS_CHAIN,
-]
```
Comment (alfetopito) [inline src/common/chains.ts:11]: This was a public export.
Won't it break anything?
Might add a deprecation warning and remove on next version?

File: src/common/index.ts [hunk 2]
```diff
@@ -1,19 +0,0 @@
-import contractNetworks from '@cowprotocol/contracts/networks.json'
-import { ALL_SUPPORTED_CHAIN_IDS } from './chains'
-
-const { GPv2Settlement } = JSON.parse(contractNetworks as unknown as string) as typeof contractNetworks
-
-export const COW_PROTOCOL_SETTLEMENT_CONTRACT_ADDRESS = ALL_SUPPORTED_CHAIN_IDS.reduce<Record<number, string>>(
```
Comment (alfetopito) [inline src/common/index.ts:6]: Is this not used externally somewhere else? Maybe mark as deprecated?

**Review by alfetopito — APPROVED** [review]

File: src/common/chains.ts (hunk 1 above)
Comment (shoom3301) [reply src/common/chains.ts:11]: It's still here, exported from `src/index.ts`

File: src/common/index.ts (hunk 2 above)
Comment (shoom3301) [reply src/common/index.ts:6]: I haven't found it used in CowSwap and Explorer both.
Maybe somebody has used it, however, they can use it from '@cowprotocol/contracts'.
I would like to discuss relations between SDK and @cowprotocol/contracts, thanks for bringing up the topic!

**Comment by bh2smith** [comment]
Anyone know why I was tagged in this?

**Review by anxolin — APPROVED** [review]

**Comment by anxolin** [comment]
> Anyone know why I was tagged in this?

@bh2smith I have no clue. I don't see you tagged. You are welcome to the party if you want

**Pull request merged**